executor/src/main/java/com/redhat/service/bridge/executor/actions/kafka/KafkaTopicActionInvoker.java

```diff
@@ -0,0 +1,48 @@
+package com.redhat.service.bridge.executor.actions.kafka;
+
+import org.eclipse.microprofile.reactive.messaging.Emitter;
+import org.eclipse.microprofile.reactive.messaging.Message;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.redhat.service.bridge.executor.actions.ActionInvoker;
+import com.redhat.service.bridge.infra.models.dto.ProcessorDTO;
+
+import io.cloudevents.CloudEvent;
+import io.smallrye.reactive.messaging.kafka.api.OutgoingKafkaRecordMetadata;
+
+public class KafkaTopicActionInvoker implements ActionInvoker {
+
+    private static final Logger LOG = LoggerFactory.getLogger(KafkaTopicActionInvoker.class);
+
+    private final String topic;
+
+    private final ProcessorDTO processor;
+
+    private final Emitter<CloudEvent> emitter;
+
+    public KafkaTopicActionInvoker(Emitter<CloudEvent> emitter, ProcessorDTO processor, String topic) {
+        this.emitter = emitter;
+        this.topic = topic;
+        this.processor = processor;
+    }
+
+    @Override
+    public void onEvent(CloudEvent cloudEvent) {
+
+        /*
+         * As the user can specify their target topic in the Action configuration, we set
+         * it in the metadata of the message we are sending.
+         * 
+         * We use the Processor id for the key on the event to Kafka to ensure ordering of events
+         * emitted by actions on this processor.
```

Comment:
If we make this choice we need to pay attention to some aspects: as far as I remember a topic can be configured to be [log compacted](https://towardsdatascience.com/log-compacted-topics-in-apache-kafka-b1aa1e4665a7) and in this case messages with same keys are removed preserving only the last occurrence (but if we control the creation of the topic this should not be critical). Moreover if a processor produces way more messages compared to the other we will have a single Kafka partition "overloaded" while other partitions are ~empty.
We can stay with this for now and maybe revisit in the future if necessary. We need to find a proper way to track similar decisions like a WIP design doc.
Wdyt?

Reply:
Thanks for the link! 

Giving it some thought, this actually feels like something we should expose as part of the Action parameters so that the user can decide what they want to do in terms of using a `key`. So I'll remove this behaviour for now so that we start with the default behaviour which is not to use a key and then gradually add the functionality when we see a need for it.

In terms of capturing how Actions will work, I'd hope that we'd start to produce documentation for Actions where we can capture behaviour like this (or show the user how to configure it).